src/modules/interest-rates/interest-rates.service.ts

```diff
@@ -1,23 +1,27 @@
-import { Injectable, InternalServerErrorException, Logger } from '@nestjs/common';
+import { Injectable, InternalServerErrorException } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
 import { InjectRepository } from '@nestjs/typeorm';
-import { DATABASE, DATE } from '@ukef/constants';
+import { DATABASE, DATE, REDACT_STRING_PATHS, REDACT_STRINGS } from '@ukef/constants';
+import { redactError } from '@ukef/helpers/redact-errors.helper';
+import { PinoLogger } from 'nestjs-pino';
 import { Equal, Repository } from 'typeorm';
 
 import { InterestRatesEntity } from './entities/interest-rate.entity';
 
 @Injectable()
 export class InterestRatesService {
-  private readonly logger = new Logger();
   constructor(
     @InjectRepository(InterestRatesEntity, DATABASE.CEDAR)
     private readonly interestRates: Repository<InterestRatesEntity>,
+    private readonly logger: PinoLogger,
+    private readonly config: ConfigService,
   ) {}
 
   findAll(): Promise<InterestRatesEntity[]> {
     try {
       return this.interestRates.find({ where: { effectiveTo: Equal(new Date(DATE.MAXIMUM_LIMIT)) } });
     } catch (err) {
-      this.logger.error(err);
+      this.logger.error(redactError(this.config.get<boolean>('app.redactLogs'), REDACT_STRING_PATHS, REDACT_STRINGS, err));
```

Comment:
Since you're using dependency injection for the logger now, we should be able to make our own logger that:
- has a PinoLogger itself (we can dependency inject it in)
- has a ConfigService (or just the `app` config) itself (we can dependency inject it in)
- has `error` and `warn` methods that take an error, call the `redactError` function on it, and then the PinoLogger's error/warn method with the result

so that we use this `redactError` function in just 1 place instead of in every service (and so we don't risk someone forgetting to use it in future)
What do you think?

Reply:
We don't use PinoLogger directly, we use nestjs module nestjs-pino as a convenient way to use PinoLogger. 
Wasn't sure what duplicating nestjs-pino was good idea, and found that PinoLogger has `hooks` that allows us redact all log messages.

Hook is defined in main.module.ts.
```
hooks: {
            logMethod(inputArgs: any, method) {
              return method.apply(this, redactStringsInLogArgs(config.get<boolean>('app.redactLogs'), REDACT_STRING_PATHS, REDACT_STRINGS, inputArgs));
            },
          },
```

Reply:
I like the `hooks` idea, great find @avaitonis!